pkg/controller/servicemesh/member/controller_test.go

```diff
@@ -173,36 +175,129 @@ func TestReconcileRemovesFinalizerFromMemberWhenMemberDeleted(t *testing.T) {
 	assert.StringArrayEmpty(updatedMember.Finalizers, "Expected finalizers list in SMM to be empty", t)
 }
 
-func TestReconcileDeletesMemberRollCreatedByItWhenMemberDeleted(t *testing.T) {
-	member := newMember()
-	member.DeletionTimestamp = &oneMinuteAgo
-	memberRoll := newMemberRoll()
-	memberRoll.Annotations = map[string]string{
-		common.CreatedByKey: controllerName,
+func TestSMMRDeletion(t *testing.T) {
+	cases := []struct {
+		name           string
+		annotations    map[string]string
+		specMembers    []string
+		otherMembers   []*maistrav1.ServiceMeshMember
+		expectDeletion bool
+	}{
+		{
+			name:           "created by user",
+			annotations:    nil, // no created-by annotation
+			expectDeletion: false,
+		},
+		{
+			name:           "created by controller, no spec.members and no other members",
+			annotations:    map[string]string{common.CreatedByKey: controllerName},
+			expectDeletion: true,
+		},
+		{
+			name:           "created by controller, members in spec.members",
+			annotations:    map[string]string{common.CreatedByKey: controllerName},
+			specMembers:    []string{"other-member"},
+			expectDeletion: false,
+		},
```

Comment:
Why the following test case below fails?
```
{
	name:           "created by controller, members in spec.members",
	annotations:    map[string]string{common.CreatedByKey: controllerName},
	specMembers:    []string{appNamespace},
	expectDeletion: true,
},
```
I expected that SMMR will be deleted if it was created by a controller and it contains the same namespace as the removed SMM.

Reply:
Because this means that some has added the namespace to `spec.members`. The controller doesn't remove the SMMR, since it has been customized by the user. When the user removes the namespace from `spec.members`, the SMMR will be deleted.